libraries/HumanPoseFuser.js

```diff
@@ -0,0 +1,803 @@
+                    // NOTE: there could be human objects in this group which are already assigned to different fused human objects
+                    if (!fusedObjectId) {
+                        fusedObjectId = fid;
+                    }
+                } else {
+                    unassignedIds.push(id);
+                }
+            }
+
+            if (fusedObjectId) {
+                // add unassigned human objects to a fused human object at the same location
+                for (let id of unassignedIds) {
+                    this.humanObjectsOfFusedObject[fusedObjectId].push(id);
+
+                    // set parent reference pointing at a fused human object
+                    this.objectsRef[id].parent = fusedObjectId;
+
+                    this.batchedUpdates[id] = {
+                        objectKey: id,
+                        frameKey: null,
+                        nodeKey: null,
+                        propertyPath: 'parent',
+                        newValue: fusedObjectId,
+                        editorId: 0    // TODO: some server identificator
+                    };
+                    if (this.verbose) {
+                        console.log('assigning human obj=' + id + ' from ' + fusedObjectId);
+                    }
+                }
+            } else {
+                // whole group is unassigned, thus create new fused human object for it
+                // TODO?: add a string identifying the server  (equivalent to globalStates.tempUuid for devices)
+                let fusedObjectId = '_HUMAN_' + 'server' + '_pose1' + utilities.uuidTime();
+                this.objectsRef[fusedObjectId] = new HumanPoseObject(this.ip, this.version, this.protocol, fusedObjectId, JOINTS);
+                if (this.humanObjectsOfFusedObject[fusedObjectId] === undefined) {
+                    this.humanObjectsOfFusedObject[fusedObjectId] = ids;
+
+                    // set parent reference pointing at new fused human object
+                    for (let id of ids) {
+                        this.objectsRef[id].parent = fusedObjectId;
+
+                        this.batchedUpdates[id] = {
+                            objectKey: id,
+                            frameKey: null,
+                            nodeKey: null,
+                            propertyPath: 'parent',
+                            newValue: fusedObjectId,
+                            editorId: 0    // TODO: some server identificator
+                        };
+                    }
+                }
+
+                // setup active heartbeat for this new object
+                server.objectBeatSender(this.beatPort, fusedObjectId, this.ip, false, true);
+
+                // add new object to different global data structures
+                this.sceneGraphRef.addObjectAndChildren(fusedObjectId, this.objectsRef[fusedObjectId]);
+                utilities.writeObject(this.objectLookupRef, this.objectsRef[fusedObjectId].name, fusedObjectId);
```

Comment:
It feels like there should be a better solution for creating a new object that's more of an all-in-one solution than all of these manual modifications and function calls but that can be left as future work for a more comprehensive server refactoring project.